src/adjoint_common.jl

```diff
@@ -492,6 +506,47 @@ function get_pf(autojacvec::EnzymeVJP; _f, isinplace, isRODE)
     end
 end
 
+function get_pf(::MooncakeVJP, prob, _f)
+    isinplace = DiffEqBase.isinplace(prob)
+    isRODE = isa(prob, RODEProblem)
+    pf = let f = _f
+        if isinplace && isRODE
+            function (out, u, _p, t, W)
+                f(out, u, _p, t, W)
+                return out
+            end
+        elseif isinplace
+            function (out, u, _p, t)
+                f(out, u, _p, t)
+                return out
+            end
+        elseif !isinplace && isRODE
+            function (out, u, _p, t, W)
+                out .= f(u, _p, t, W)
+                return out
+            end
+        else
+            # !isinplace
+            function (out, u, _p, t)
+                out .= f(u, _p, t)
+                return out
```

Comment:
why force a return of `out` instead of `nothing`?

Reply:
This is so that I can make use of Mooncake's `__value_and_pullback!!` function. I might make a non-breaking PR in future if this changes.

Reply:
(I agree that it's a bit unfortunate that I'm doing this though -- if I made use of lower-level Mooncake interfaces I probably wouldn't have to do it, but this way saves on a lot of code)